### DISCUSSION TITLE
Amiga support

### DISCUSSION THREAD

**Discussion by ozkanocakli** [erichelgeson/BlueSCSI-v1#34] [General]
bluscsi's current firmware amiga hdtoolbox recognize bluscsi but when try to format harddrive bluesci hdd activily led turn on solid and wait

**Comment by erichelgeson** [comment]
Thanks for the report. No "core" developers have Amiga's so will compatibility likely be hit or miss. I know some have reported success but it seems to depend on system/card/etc.

**Reply by C128Guy** [reply]
Any update on this with Amiga's?  I have one here with a GVP controller locking up at the same point.  There was a youtube video saying they made changes to the firmware but not specific enough.  Anyone have working firmware I can use with the Amiga?

**Reply by Retro-updates** [reply]
I have tried an updated firmware but with same result. On my GVP SCSI in A2000 i do see the BlueSCSI disks but cannot save the config in HDToolbox. On my A500 with DataFlyer SCSI the BlueSCSI does not show up.
Hoping that there will be a fix as currently the Alternative RASCSI is problematic because of difficulties in getting Raspberry PI's

**Reply by C128Guy** [reply]
Did you see the video on youtube? A delay change is mentioned in the code but doesn't go into detail how he got it working.  [https://www.youtube.com/watch?v=RyzWCMf_OXo&ab_channel=GouldFishOnGames](url)

**Reply by C128Guy** [reply]
We need something for Amiga....SCSI2SD are backordered now too!

**Reply by Retro-updates** [reply]
agree. i'm hoping also for a fix for BlueSCSI. Seems a great product but now mostly for Macintosh.
I've contacted the guy from the youtube item you mention, but he did not have that information anymore.
From discord i got information about the new firmware. i did try it but with same result.
This is the firmware i've tried.
Maybe you have more luck ;-)

[TESTfirmware.zip](https://github.com/erichelgeson/BlueSCSI/files/8381264/TESTfirmware.zip)

**Reply by erichelgeson** [reply]
Hi all, There has been quite a few changes to the main branch since this last conversation, which may help with Amiga (or may not). I do hope to have a beta firmware out sometime this week/weekend that hopefully an Amiga user can try out and report back.

**Reply by C128Guy** [reply]
I've got an Amiga sitting here waiting :)

**Reply by Retro-updates** [reply]
Thanks. Will for sure test them out and report back the results!

**Reply by idrougge** [reply]
For what it's worth, the original ArdSCSIno-STM32 works on Amiga, so if BlueSCSI doesn't, that is something introduced by the fork.

**Reply by C128Guy** [reply]
SCSI2SD works well, I have it in an A2000 and get 2mb/sec transfers.  Just none to be had at the moment!

**Comment by prickle** [comment]
I also tried on an Amiga. In fact, I tried a few systems.

*Amiga 500 with A590 expansion - in HDToolbox, bluescsi does not show up at all
*Sun Sparcstation 10 - using probe-scsi, bluescsi does not show up at all
*486 pc with Adaptec AHA-1542 - in it's BIOS bluescsi is seen, but "Command not supported" errors for format or verify

As these are the only SCSI systems I have, I can't be sure the bluescsi itself is working ok. I guess I will try to turn the serial debugging on and see what it has to say for itself.

**Reply by ea4gge** [reply]
It may need Eprom update. Here, in this video, my friend Armand install one of the BlueSCSI's i did in a A590:

https://www.youtube.com/watch?v=pbIozWBvdXw

I am also in that video giving an explaination about

**Comment by prickle** [comment]
So, on the AHA-1542
verify:
`Command:2F:0:0:0:0:0:0:8:0:0:ID 1:LUN 0
[*Unknown]`
format:
`Command:15:10:0:0:C:0:ID 1:LUN 0
[*Unknown]`

On the Amiga
during boot:
`Command:80:8:0:0:x:1:ID 1:LUN 0
[*Unknown]`
16 times where x increments from 0 to F

This time the drive showed up in HDtoolbox. When starting up it sends these commands:
`Selection
MsgOut2
Command:80:12:0:0:0:24:ID 1:LUN 0
[*Unknown]
Sts
MsgIn
BusFree
Selection
MsgOut2
Command:80:37:0:8:0:0:ID 1:LUN 1
[*Unknown]
Sts
MsgIn
BusFree
Selection
MsgOut2
Command:80:37:0:8:0:0:ID 1:LUN 1
[*Unknown]
Sts
MsgIn
BusFree
`
Trying to ID the drive sends these messages:
`Selection
MsgOut2
Command:80:1A:0:3:0:FF:ID 1:LUN 0
[*Unknown]
Sts
MsgIn
BusFree
Selection
MsgOut2
Command:80:12:0:0:0:24:ID 1:LUN 0
[*Unknown]
Sts
MsgIn
BusFree
`
and responds "Insufficient data returned by inquiry".

**Comment by prickle** [comment]
Now, after fiddling (removed a previous debug attempt), the amiga does the following on boot:
`BusReset!
BusFree
Selection
MsgOut2
Command:12:0:0:0:FE:0:ID 1:LUN 0
[Inquiry]
DATAIN PHASE`
and hangs there, waiting for something?

**Comment by prickle** [comment]
Adding some debug prints like this:
```
inline void writeHandshake(byte d)
{
  GPIOB->regs->BSRR = db_bsrr[d]; // setup DB,DBP (160ns)
  SCSI_DB_OUTPUT() // (180ns)
  // ACK.Fall to DB output delay 100ns(MAX)  (DTC-510B)
  SCSI_OUT(vREQ,inactive) // setup wait (30ns)
  SCSI_OUT(vREQ,inactive) // setup wait (30ns)
  SCSI_OUT(vREQ,inactive) // setup wait (30ns)
  SCSI_OUT(vREQ,active)   // (30ns)
  //while(!SCSI_IN(vACK)) { if(m_isBusReset){ SCSI_DB_INPUT() return; }}
  Serial.println(" -- vREQ Active");
  while(!m_isBusReset && !SCSI_IN(vACK));
  Serial.println(" -- got vACK");
  // ACK.Fall to REQ.Raise delay 500ns(typ.) (DTC-510B)
  GPIOB->regs->BSRR = DBP(0xff);  // DB=0xFF , SCSI_OUT(vREQ,inactive)
  // REQ.Raise to DB hold time 0ns
  SCSI_DB_INPUT() // (150ns)
  while( SCSI_IN(vACK)) { if(m_isBusReset) return; }
  Serial.println(" -- release vACK");
}
```
gives me this:
```
BusReset!
Selection
MsgOut2
Command:12:0:0:0:FE:0:ID 1:LUN 0
[Inquiry]
DATAIN PHASE
Writing 36 Bytes.
 -- vREQ Active
 -- got vACK
 -- release vACK
 -- vREQ Active
 -- got vACK
 -- release vACK
 -- vREQ Active
 -- got vACK
 -- release vACK
 -- vREQ Active
 -- got vACK
 -- release vACK
 -- vREQ Active
 -- got vACK
 -- release vACK
 -- vREQ Active
 -- got vACK
 -- release vACK
 -- vREQ Active
```
Seems like six bytes are sent, then hangs waiting for vACK. My guess is the controller misses the first byte of the response then misinterprets the additional data length as zero, hence it's expecting a total message length of six and refuses to take any more. Maybe?

**Reply by prickle** [reply]
Line breaks in code blocks not working for me? It's not my day.

**Comment by ozkanocakli** [comment]
In my setup, a590 recognise on hdtoolbox, I setup partions and it writes data, then I try to format drive, but it stucks on process

**Reply by prickle** [reply]
You get further than me although our setups should be very similar. What could be different? Some edge case timing issues? Floating lines? Perhaps I am using a poor quality STM clone board.

**Comment by ea4gge** [comment]
Hi, all.

I have already working the BlueSCSI in an Amiga CDTV. it is working with the CDTV SCSI interface, 2MB of CHIP ram, 8 MB of Fast ram and Kick/WB 3.2. Basically the CDTV SCSI device is an A590/A2090 card. The code was uploaded to the BluePill with USB to serial interface and the Arduino IDE (code without modiffications) and the only issue i has for to start was the SCSI's terminations. I hast to test some configurations but... finally... Controller withouth terminators, JP1 not shorted, JP4 and 5 shorted, JP9 shorted, and blueSCSI powered by USB (i have to test another config but that is at the moment).

The img used for the test is a 100MB img named as: HD10_100.hda, partitioned with the HDToolBox (yes, is directly recogniced).


Edit:

The log file from the SD:

BlueSCSI <-> SD - https://github.com/erichelgeson/BlueSCSI
VERSION: 1.1-SNAPSHOT
DEBUG:0 SCSI_SELECT:0 SDFAT_FILE_TYPE:3
SdFat version: 2.1.0
SdFat Max FileName Length: 32
Initialized SD Card - lets go!
Sd MID:2 OID:TM
Sd Name:SD01G
Sd Date:11/20010
Sd Serial:2711534736
Not an image: LOG.txt
Imagefile: HD10_100.hda / 94371840bytes / 92160KiB / 90MiB
!! More than one image did not specify a SCSI ID. Last file will be used at ID 1. !!
ID:LUN0:LUN1:
 0:----:----:
 1: 512:----:
 2:----:----:
 3:----:----:
 4:----:----:
 5:----:----:
 6:----:----:
Finished initialization of SCSI Devices - Entering main loop.

**Comment by ea4gge** [comment]
Ok... now i have correctly renamed the IMG file to **HD00_512.hda**. The log:

BlueSCSI <-> SD - https://github.com/erichelgeson/BlueSCSI
VERSION: 1.1-SNAPSHOT
DEBUG:0 SCSI_SELECT:0 SDFAT_FILE_TYPE:3
SdFat version: 2.1.0
SdFat Max FileName Length: 32
Initialized SD Card - lets go!
Sd MID:2 OID:TM
Sd Name:SD01G
Sd Date:11/20010
Sd Serial:2711534736
Not an image: LOG.txt
Imagefile: HD00_512.hda / 94371840bytes / 92160KiB / 90MiB
!! More than one image did not specify a SCSI ID. Last file will be used at ID 1. !!
ID:LUN0:LUN1:
 0: 512:----:
 1:----:----:
 2:----:----:
 3:----:----:
 4:----:----:
 5:----:----:
 6:----:----:
Finished initialization of SCSI Devices - Entering main loop.

**Reply by ozkanocakli** [reply]
I removed termination jumpers, and power via usb port, A590 hangs on booting

**Reply by ea4gge** [reply]
yes, same here. Termination jumpers closed, J9 closed, powered by USB and the terminations of the controler extracted (in my case) with this config is working

**Reply by ozkanocakli** [reply]
could you please send picture of your current configuration

**Reply by ea4gge** [reply]
yes, of course:

![photo_2021-09-14_00-22-40](https://user-images.githubusercontent.com/90650383/133164353-7c2dd279-6302-40ab-bcfd-3f1db060033d.jpg)

Controller and BluSCSI. Sorry, i forget to menction this point: the board is not the same, so the jumper's name will be different. In my case i did the board of:

https://github.com/xunker/bluescsi_pb

Basically the difference is the dual connectors for SCSI so it must to work equal. The terminators are 330 Ohms and 220 Ohms. The terminators of the SCSI CDTV's card are the same, 330 and 220. In all my test when terminators are not correct the isue is exactly the same as you have descrived:

**"I removed termination jumpers, and power via usb port, A590 hangs on booting"**

So... Can be in your card the terminators values the issue? i has before, in the SD2SCSI previouslly installed, the terminators of the design: 100 ohms,  and the SCSI card's terminator installed also and it was working fine but not in this case. The only way for to have it working was the described  : SCSI card without terminators and BlueSCSI with them activated.

**Reply by ozkanocakli** [reply]
both two boards are different from mine, my terminators values are 220 ohm and 390 ohm, so i have to find 330 ohm and test it again, thanx for pictures and feedback

**Reply by ea4gge** [reply]
Thanks to you. I'll be waiting your update, i hope this comments help you

**Comment by Retro-updates** [comment]
Anyone more luck in getting the BlueSCSI working with an Amiga SCSI controller? I tried the latest build v1.1-20211215 with a BlueSCSI 1.1-a rev board on a GVP SCSI controller. HDToolbox does find the virtual disks but cannot configure them.

**Reply by ea4gge** [reply]
In GVP i have not idea. in CDTV, A590, A4000, A2090/2091, A1200 with blizzard and A3000 is working without issues already.

BR. ea4gge/// Alberto

**Reply by andabata** [reply]
I got it semi working on a G-Force 030 Combo, when i created an image with UAE and used that, it booted. If i tried writing to it, it hangs.

**Reply by ea4gge** [reply]
Strage... write protection, may be?

**Reply by andabata** [reply]
That would be to easy :), I used it on a tekram controller (pc) aswell. and there it was fully functional. On an adaptec controller it failed completly.

**Comment by Retro-updates** [comment]
Did you use any specific virtual image file? Are you using standard bluescsi software?

**Reply by ea4gge** [reply]
V1.1 of the firm
The IMG must to have 512 bytes per sector

Only that is needed. At the moment i have 6 IMG mounted as the same time as 6 diferent SCSI units in the same BlueSCSI and with different partitions. Yes, it can mount diferent IMG at the same time

**Comment by ea4gge** [comment]
Ah!! another think for the GVP:

The GVP for the A500 have not Power Supply in the SCSI connector so... You must to modify it or give Power to the BluePill in the USB connector. I don't know in the case of the GVP for A2000. In that case is important to config the jumpers on the blueSCSI for isolate the PS pin on the SCSI connector. In the GVP this pin is GND

**Reply by Retro-updates** [reply]
I did provide power through the berg connector on both the A500 - dataflyer SCSI set-up and the A2000 GVP SCSI set-up. The A2000 with GVP does see the images on the BlueSCSI microsd but i cannot save them in HDToolbox after i try to create a partition.

**Reply by ea4gge** [reply]
what error you have? what version of the Amiga OS?

**Reply by Retro-updates** [reply]
I've tested on 3.2

**Reply by Retro-updates** [reply]
not sure what the exact error was, but something that it cannot write configuration.
I can read configuration and it gives me the correct disk information (In my case Quantum fireball 1GB), but saving the config gives this error GVP with A2000 / KS 3.2)

**Reply by C128Guy** [reply]
I get the same error writing config. It does not matter which version of the KS rom.

**Reply by ea4gge** [reply]
sure, i have it working with ROMs 3.1 and 3.2 without issues. Is for to try to identify with the error the possible problem

**Reply by Retro-updates** [reply]
i will set-up again and will post te exact error here with picture ok?

**Reply by ea4gge** [reply]
OK, perfect

Check this video:

https://www.youtube.com/watch?v=pbIozWBvdXw

I posted it in a message before. Is a video from Armand, a friend, and me giving an explaination with my CDTV. In this video you can see how to install in a A590

Is in Spanish, sorry

**Reply by Retro-updates** [reply]
i will view it (and try to understand the Spanish ;-)

**Reply by ea4gge** [reply]
HIHIHI... MAy be subtitles :p

**Reply by Retro-updates** [reply]
Just switched on the A2000 with BlueSCSI connected. (Image set as ID1).
Both internal SCSI disk and BlueSCSI ID1 appear.

Attached images of the error message (when trying to save config):
Just to complete: pressing read configuration first time gave the Quantum fireball details.
When pressing the OK button, the error comes up.

![IMG_6637](https://user-images.githubusercontent.com/82937588/160867554-1edf11a2-13b7-48f4-a54a-8d8767486549.jpg)
![IMG_6636](https://user-images.githubusercontent.com/82937588/160867627-ed909da5-7637-45bd-9ec7-dd48b7402131.jpg)

Hope it helps to identify the problem.

**Reply by Retro-updates** [reply]
> OK, perfect
> 
> Check this video:
> 
> https://www.youtube.com/watch?v=pbIozWBvdXw
> 
> I posted it in a message before. Is a video from Armand, a friend, and me giving an explaination with my CDTV. In this video you can see how to install in a A590
> 
> Is in Spanish, sorry

View the video, but besides from the Spanish ;-), i cannot understand what was done to get this to work.
I do see that he's using the PowerBook adapter version of bluescsi
I'm using the BleuSCSI 1.1-a board

**Reply by C128Guy** [reply]
Made a bit of progress.  Pulled out an old 2091 card.  It took the hd setup and I installed an OS in winUAE.  It almost boots.  Gets to the WB screen and locks up.  I assume it is trying to write data to the drive at the point or something.

**Reply by Retro-updates** [reply]
so you configured an image with WinUAE and used that as image on the MicroSD Card?
Did you add the SCSI drivers as well?

**Reply by C128Guy** [reply]
Yes.  But found my problem, I tried to run 2.1 on a 2.04 rom.  Reinstalled and it's actually working with the 2091.  Now to get the GVP to work since it has 8mb or ram.

**Reply by Retro-updates** [reply]
The thing is that commodore Amiga has quite a number of different SCSI controllers, although GVP is one of the most common and known one. I have one in my A2000, but a DataFlyer 500 for my A500.

**Reply by idrougge** [reply]
2.1 always runs on a 2.04 ROM.

**Reply by C128Guy** [reply]
That's what I thought, but a reinstall of 2.04 worked.

**Reply by C128Guy** [reply]
What is also strange - the 2.04 version of hdtoolbox couldn't see the drive, but 2.1 does.

**Reply by Retro-updates** [reply]
And both have the same SCSI driver?

**Reply by C128Guy** [reply]
No.  GVP card uses gvpscsi.device vs scsi.device

Or you asking about WB versions?  That I don't know.

**Reply by idrougge** [reply]
So it’s not just that HDToolbox’s tooltype pointed to the wrong scsi.device?

**Reply by Retro-updates** [reply]
I can imagine that gvpscsi.device is bit different than scsi.device but i might be wrong

**Reply by Retro-updates** [reply]
> So it’s not just that HDToolbox’s tooltype pointed to the wrong scsi.device?

During start-up of HDToolbox i see HDToolbox scanning devices on gvpscsi.device, finding both internal SCSi disk and the virtual BlueSCSI disk on correct ID (as set in name of image)

**Reply by C128Guy** [reply]
Its more than just a driver issue with the GVP.  I just tried my working image on the GVP card and it won't even see the drive to boot.

**Reply by Retro-updates** [reply]
I would think the same yes. The GVP driver does see the BlueSCSI devices on ID scan but then stops working. Hope that one of the software developers of BluesCSI firmware either has an Amiga or can think of a solution to test.

**Reply by erichelgeson** [reply]
I have published a new pre-release .bin. No guarantees it changes or fixes anything for Amiga, but something to try. 

Please continue to report any Amiga issues, troubleshooting, discussions in this thread vs the release thread. 

https://github.com/erichelgeson/BlueSCSI/releases/tag/v1.1-BETA-20220330

**Reply by Retro-updates** [reply]
Thanks. Will let know the test results.

**Reply by Retro-updates** [reply]
Tried the firmware v1.1-BETA-20220330 but with same error when trying to save configuration from HDtoolbox.

BlueSCSI <-> SD - https://github.com/erichelgeson/BlueSCSI
VERSION: 1.1-BETA-20220330
DEBUG:0 SCSI_SELECT:0 SDFAT_FILE_TYPE:3
SdFat version: 2.0.6
SPI speed: 50Mhz
SdFat Max FileName Length: 32
Initialized SD Card - lets go!
Sd MID:3 OID:SD
Sd Name:SD64G
Sd Date:12/2014
Sd Serial:3642636730
 - HD10_512_50M_AMIGA.hda / 52428800bytes / 51200KiB / 50MiB
ID:LUN0:LUN1:
 0:----:----:
 1: 512:----:
 2:----:----:
 3:----:----:
 4:----:----:
 5:----:----:
 6:----:----:
Finished initialization of SCSI Devices - Entering main loop.

Not sure if it is of any help, but here's the log from the MicroSD

**Reply by ea4gge** [reply]
HD10_512_50M_AMIGA.hda --> This name of file configure the image at HD 1:

0:----:----:
1: 512:----:
2:----:----:
3:----:----:
4:----:----:
5:----:----:
6:----:----:

If you want to config as HD0 the name must to be:

HD00_512_50M_AMIGA.hda

**Reply by Retro-updates** [reply]
i am aware of this. I'm testing the BlueSCSI next to an internal SCSI disk that is on ID0.
This is where my original Amiga Workbench now boots from.
Then i go to HDToolbox to try to create a partition on the second (ID1) BlueSCSI disk.

**Reply by ea4gge** [reply]
OK. Have you tested the BlueSCSI as only disk before?

**Reply by Retro-updates** [reply]
Not yet, but i can try of course if you think it could make a difference,

**Reply by ea4gge** [reply]
may be, because the terminators. Only for test

**Reply by Retro-updates** [reply]
was away for few days, but will try to test again in the next few days

**Reply by Retro-updates** [reply]
Tested with BlueSCSI (ID0) as only drive on the GVP SCSI controller.
Booted from AmigaOS3.2 install disk, HDTooldBox scanned for SCSI devices on the GVP scsi device and found the virtual disk on the BlueSCSI.
I was able to read and save the disk configuration. 
HDTooldbox showed the Quantum fireball and i was able to partition the drive.

BUT: when i try to save changes to drive and exit HDToolbox, the system seems to freeze (Mouse moves and UI Buttons seem to respond (highlight when clicked), but i cannot Save the changes to drive and cannot Exit HDToolbox.

When simply rebooting the Amiga (2000) and try to install AmigaOS3.2 there is no harddrive found (because i could not save the changes to drive after partitioning).

**Reply by Retro-updates** [reply]
Today ive'tried the BlueSCSI beta firmware (BlueSCSI-1.1-BETA-20220330.bin) on my A500 with DataFlyer 500 SCSI controller and THAT WORKED!
Booted from the DataFlyer (3.7e) diskprep diskette and then the virtual drive on the BlueSCSI MicroSD was found.
I was able to partition it and able to install AmigaOS3.2 on it without any issues.

It seems that BlueSCSI firmware is starting to be compatible with more Amiga SCSI controllers!

![IMG_6739](https://user-images.githubusercontent.com/82937588/162619741-0671a970-2085-4e0c-9183-371c45ad00e3.jpg)

**Comment by marcelv-3** [comment]
for the A2000 with GVP, the only way I am able to create partitions and format the drive is to set debug = 2 in bluescsi, upload the image and start the A2000, using faaastPrep to partition/format
 
Debug = 2 logs to the sd card, very slow, so the problem muat be a timing issue,

After partitioning/formatting you can set debug = 0 and use that image to run the Amiga

can also upload an gvp prepped hda image somewhere

**Reply by ozkanocakli** [reply]
Hi, could you please upload the image file or send email

**Reply by marcelv-3** [reply]
[HD00_512-WB2.zip](https://github.com/erichelgeson/BlueSCSI/files/8923203/HD00_512-WB2.zip)

**Reply by marcelv-3** [reply]
let me know if the hdd image works,
i changed the bluescsi code a little to make it more Amiga compatible so you may need/want that too,

**Reply by ea4gge** [reply]
What did you changed in the code? i have using the 1.1 version before the compilation, i don't know if it have any different with the last uploaded, and for me and other friends, it is working fine on Amiga:


/*  
 *  BlueSCSI
 *  Copyright (c) 2021  Eric Helgeson, Androda
 *  
 *  This file is free software: you may copy, redistribute and/or modify it  
 *  under the terms of the GNU General Public License as published by the  
 *  Free Software Foundation, either version 2 of the License, or (at your  
 *  option) any later version.  
 *  
 *  This file is distributed in the hope that it will be useful, but  
 *  WITHOUT ANY WARRANTY; without even the implied warranty of  
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU  
 *  General Public License for more details.  
 *  
 *  You should have received a copy of the GNU General Public License  
 *  along with this program.  If not, see https://github.com/erichelgeson/bluescsi.  
 *  
 * This file incorporates work covered by the following copyright and  
 * permission notice:  
 *  
 *     Copyright (c) 2019 komatsu   
 *  
 *     Permission to use, copy, modify, and/or distribute this software  
 *     for any purpose with or without fee is hereby granted, provided  
 *     that the above copyright notice and this permission notice appear  
 *     in all copies.  
 *  
 *     THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL  
 *     WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED  
 *     WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE  
 *     AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR  
 *     CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS  
 *     OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,  
 *     NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN  
 *     CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.  
 */

// Log File
#define VERSION "1.1-SNAPSHOT"
#define LOG_FILENAME "LOG.txt"
FsFile LOG_FILE;



I have uploaded this directly by RS232 to the BluePill from the Arduino IDE. It is working fine in my CDTV

**Reply by ozkanocakli** [reply]
Thanks for rhe image

**Reply by Retro-updates** [reply]
Did you use the standard BlueSCSI-1.1-20211215.bin firmware ?

**Reply by marcelv-3** [reply]
used the 1.1-SNAPSHOT-20220522
will try later with the latest snapshot

**Reply by ozkanocakli** [reply]
I tried your image file, a590 it boots, unfortunately i can write any file to hdd, it hangs and corrupt image file

my config
A500
kick3.2
A590 scsi controller

blue retro
firmware BlueSCSI-1.1-20220404
usb powered bluepill

**Reply by ea4gge** [reply]
A590 must to be updated in the ROM side

[A590_2091_PDMA07_Roms.zip](https://github.com/erichelgeson/BlueSCSI/files/8938920/A590_2091_PDMA07_Roms.zip)


Atached lastest ROMS

**Reply by ea4gge** [reply]
> Did you use the standard BlueSCSI-1.1-20211215.bin firmware ?

I have used the "before compilation"(20210813 cpp file ported to .ino file) 1.1 one (attached). In my case uploaded by arduino IDE RS232 option

[BlueSCSI_1-1.zip](https://github.com/erichelgeson/BlueSCSI/files/8938992/BlueSCSI_1-1.zip)

If have not changes

**Comment by marcelv-3** [comment]
Tested also with latest snapshot build 20220617, same problem still exists as in the previous snapshot and the latest build, unable to format/write unless I set debug to 2 (write to file), makes things very slow but everything works, format, read, write,..
In last snapshot the code to log to file is removed so had to put this back.  

The problem is in the Data out phase. / readdataloop. 
Maybe a timing problem.?

Seems like a more general problem for some non macintosh scsi controllers.

@erichelgeson How to best debug this? Can provide a log with complete boot/read/write (debug=2)

testeds with config:
A2000
kick 2
GVP scsi controller

**Reply by ea4gge** [reply]
> Tested also with latest snapshot build 20220617, same problem still exists as in the previous snapshot and the latest build, unable to format/write unless I set debug to 2 (write to file), makes things very slow but everything works, format, read, write,.. In last snapshot the code to log to file is removed so had to put this back.
> 
> The problem is in the Data out phase. / readdataloop. Maybe a timing problem.?
> 
> Seems like a more general problem for some non macintosh scsi controllers.
> 
> @erichelgeson How to best debug this? Can provide a log with complete boot/read/write (debug=2)
> 
> testeds with config: A2000 kick 2 GVP scsi controller

May be but as we have tested... in A4000 (pending to confirm the controller), A590 (ROMs updated), A2091 (ROMs updated), A1200 (Blizzard IV) and CDTV are working fine so... i guess is an issue with the GVP

**Comment by marcelv-3** [comment]
Tried your "before compilation"(20210813 cpp file ported to .ino file) 1.1 
Same results. Hangs on write or gives write error on the first block it is supposed to write.
Am glad it is working on other Amiga/SCSI combinations but want to make this working. Is working fine with a real HD and with debug on, so has to be a software issue.
The ROM (gvpscsi.ddevice) is the latest  version 3.15. There is a unofficial 4.15 version so might try that, but as said earlier, the problem occurs on other systems too (samplers / sun, .. see other discussions)

**Comment by marcelv-3** [comment]
Write to hd is working using the code from the following pull request:
https://github.com/erichelgeson/BlueSCSI/pull/142

@ozkanocakli Could you try this version to see if this also fixes your write to hd problem.

**Reply by ozkanocakli** [reply]
could you please upload firmware, i could not see it on request 142

**Reply by marcelv-3** [reply]
will upload tonight

**Reply by marcelv-3** [reply]
firmware and HD image to test, not an official release, only to check if this solves the write problem

[fwpull142.zip](https://github.com/erichelgeson/BlueSCSI/files/9115658/fwpull142.zip)
[iHD00_512-E.zip](https://github.com/erichelgeson/BlueSCSI/files/9115659/iHD00_512-E.zip)

**Comment by ozkanocakli** [comment]
on stm32 boards still gives read/write errors , on ch32 working, could you please apply this fix to latest firmware that supports cd images

**Comment by nd-mk** [comment]
Hi,

can someone help me?
I bought some bluescsi cards that were supposed to work with the Amiga, but both on Oktagon 2008 and GVP controllers don't work ... there are four identical cards and I've been trying all day and I can't get them to work
The problem is that they don't allow booting my A2000s, the system hangs, not even allowing booting from the floppy disk
If I then reconnect the previous normal SCSI HDs everything works perfectly
What should I do ?
The LOG file is generated, but then everything freezes, as if these bluescsi drives are incompatible with the SCSI controllers
If someone is kind enough to help me, otherwise I'll send them back to the sender or throw them out the window...

this is an example of LOG that is generated in the root of the SD, as well as many other HD files that I have tried and none have solved the problem

BlueSCSI <-> SD - https://github.com/erichelgeson/BlueSCSI
VERSION: 1.1-SNAPSHOT-20220627
DEBUG:0 SDFAT_FILE_TYPE:3
SdFat version: 2.1.2
Sd Format: FAT32FAT16 - Consider formatting the SD Card with exFAT for improved performance.
SPI speed: 50Mhz
SdFat Max FileName Length: 32
Initialized SD Card - lets go!
Sd MID:9F OID:TI
SD Name:SD32G
Sd Date: 1/2015
SD Serial:3943403648
 - HD00_512.hda / 524288000bytes / 512000KiB / 500MiB
ID:MON0:
 0:512:
 1:----:
 2:----:
 3:----:
 4:----:
 5:----:
 6:----:
Finished initialization of SCSI Devices - Entering main loop.

OR here an other exemple:

BlueSCSI <-> SD - https://github.com/erichelgeson/BlueSCSI
VERSION: 1.1-SNAPSHOT-20220627
DEBUG:0 SDFAT_FILE_TYPE:3
SdFat version: 2.1.2
Sd Format: FAT32FAT16 - Consider formatting the SD Card with exFAT for improved performance.
SPI speed: 50Mhz
SdFat Max FileName Length: 32
Initialized SD Card - lets go!
Sd MID:9F OID:TI
Sd Name:SD32G
Sd Date:1/2015
Sd Serial:3943403648
 - HD00_512-E.hda / 42991616bytes / 41984KiB / 41MiB
ID:LUN0:
 0: 512:
 1:----:
 2:----:
 3:----:
 4:----:
 5:----:
 6:----:
Finished initialization of SCSI Devices - Entering main loop.

Oktagon 2008 and GVP controllers are terminated by default, and all tried bluescsi are also terminated, what should I do?

thanks to who helps me

**Comment by marcelv-3** [comment]
You are using software VERSION: 1.1-SNAPSHOT-20220627
Could you update to the latest version? v1.1-20220917
The latest version works much better in an Amiga with the GVP controller.

You can also try my hdd image a few posts back (iHD00_512-E.zip)

**Reply by C128Guy** [reply]
I was able to get it to work with the HC+8 card but it was not easy.  First I had to initialize the disk with the Amiga, but couldn't get the o/s to install.  Then I put the card into my laptop and install the o/s using WinUAE.  Only way I could get it to work.  

I got absolutely no where with the A2091 card.

**Comment by nd-mk** [comment]
hi

yes, It's 1.1-SNAPSHOT-20220627
Could you please upload the firmware v1.1-20220917?

could you also please point me to a simple programmer for a few € that I can order from aliexpress to reprogram it?

one last question, did you have to remove the termination resistors in your GVP controller to get your bluescsi card to work properly? or did you have to remove the termination jumpers from the bluescsi card?

Thanks for your help

**Comment by marcelv-3** [comment]
https://github.com/erichelgeson/BlueSCSI/releases/tag/v1.1-20220917
download links at bottom of page

termination on GVP and BlueScsi are on 

For the programmer I use a ST-Link V2, not from ali but from a local store. 
Where are you located?

**Comment by ozkanocakli** [comment]
selamlar

samba yı enable etmem rağmen, yazılım cihazı bulamıyor, fsg-3 ün default ip
adresi ve password u nedir

iyi günler
